src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -647,27 +651,34 @@ def handleEnforceGUIDInFileName(self):
         if self.crabPSet:
             return
 
-        # only check this if we are in the first step of a workflow as output files of chained processing
-        # steps will not have correct GUID information
-        if self.step.data._internal_name != "cmsRun1":
-            self.logger.info("Not evaluating enforceGUIDInFileName parameter for step %s",
-                             self.step.data._internal_name)
-            return
+        if secondaryInput:
+            inputSource = secondaryInput
+            self.logger.info("Evaluating enforceGUIDInFileName parameter for secondary input data.")
+        else:
+            inputSource = self.process.source
 
         # only enable if source is PoolSource or EmbeddedRootSource
-        if self.process.source.type_() not in ["PoolSource", "EmbeddedRootSource"]:
+        if inputSource.type_() not in ["PoolSource", "EmbeddedRootSource"]:
             self.logger.info("Not evaluating enforceGUIDInFileName parameter for process source %s",
-                             self.process.source.type_())
+                             inputSource.type_())
             return
 
         self.logger.info("Evaluating if release %s supports enforceGUIDInFileName parameter...",
                          self.getCmsswVersion())
 
         # enable if release supports enforceGUIDInFileName and parameter is not set
         if (isEnforceGUIDInFileNameSupported(self.getCmsswVersion()) and
-                not hasattr(self.process.source, "enforceGUIDInFileName")):
+                not hasattr(inputSource, "enforceGUIDInFileName")):
```

Comment:
I'm not sure whether we need this `not hasattr`. Which means, even if it comes set from the PdmV/PPD side, we should be able to override it if needed.